cmplibrary/src/main/java/com/sourcepoint/cmplibrary/SpConsentLibImpl.kt

```diff
@@ -208,6 +209,19 @@ internal class SpConsentLibImpl(
         }
     }
 
+    override fun customConsentGDPR(vendors: Array<String>, categories: Array<String>, legIntCategories: Array<String>, successCallback: CustomConsentClient) {
+        customConsentGDPR(
+            vendors = vendors.toList(),
+            categories = categories.toList(),
+            legIntCategories = legIntCategories.toList(),
+            success = {
+                it?.let { spc ->
+                    successCallback.transferCustomConsentToUnity(spc.toJsonObject().toString())
+                } ?: run { spClient.onError(RuntimeException("An error occurred during the custom consent request: SPConsent is null")) }
```

Comment:
ok, if you say "An error occurred during the custom consent request" and then "SPConsent is null" means that the error is that `SPConsent` is null but "SPConsent is null" is not the cause, it is the effect (an error occurred so you get a null object).